Source/OpenDRIVEEditor/Private/OpenDriveRoadEd.cpp

```diff
@@ -0,0 +1,279 @@
+			SetArrowMeshes(laneSpline, mesh, false);
+		}
+		else
+		{
+			SetArrowMeshes(laneSpline, mesh, true);
+		}
+	}
+	//if its not a junction, we draw the spline meshes, else we just keep the spline line 
+	if (_junctionId == -1)
+	{
+		// Spline meshes creation 
+		SetColoredLaneMeshes(laneSpline);
+	}
+	else
+	{
+		//set driving lane's spline's color in junction
+		switch (_lane->GetLaneType())
+		{
+		case (roadmanager::Lane::LaneType::LANE_TYPE_DRIVING):
+			laneSpline->EditorUnselectedSplineSegmentColor = _lane->GetId() > 0 ? FLinearColor::Red : FLinearColor::Green;
+			break;
+
+		case(roadmanager::Lane::LaneType::LANE_TYPE_RESTRICTED):
+			laneSpline->EditorUnselectedSplineSegmentColor = FLinearColor(1, 0.5, 0);
+			break;
+
+		case(roadmanager::Lane::LaneType::LANE_TYPE_BIKING):
+			laneSpline->EditorUnselectedSplineSegmentColor = FLinearColor(0, 0, 1);
+			break;
+		}
+	}
+}
+
+void AOpenDriveRoadEd::SetLanePoint(USplineComponent* laneSpline_, roadmanager::Position& position, double s, float offset)
+{
+	position.SetLanePos(_roadId, _lane->GetId(), s, 0.);
+	
+	FVector sp;
+	sp = CoordTranslate::OdrToUe::ToLocation(position);
+	sp.Z += offset;
+	laneSpline_->AddSplineWorldPoint(sp);
+	FRotator rot = FRotator::ZeroRotator;
+	rot.Yaw = -FMath::RadiansToDegrees(position.GetH());
+	laneSpline_->SetRotationAtSplinePoint(laneSpline_->GetNumberOfSplinePoints() - 1, rot, ESplineCoordinateSpace::World);
+	float Yscale = (_laneSection->GetWidth(position.GetS(), _lane->GetId()) * 100) / _baseMeshSize;
+	laneSpline_->SetScaleAtSplinePoint(laneSpline_->GetNumberOfSplinePoints() - 1, FVector(1.0f, Yscale, 1.0f));
+}
+
+void AOpenDriveRoadEd::CheckLastTwoPointsDistance(USplineComponent* laneSpline_, float step_)
+{
+	float dist = FVector::Distance(laneSpline_->GetWorldLocationAtSplinePoint(laneSpline_->GetNumberOfSplinePoints() - 2),
+		laneSpline_->GetWorldLocationAtSplinePoint(laneSpline_->GetNumberOfSplinePoints() - 1));
+	if ((dist / 100) < step_ / 3)
+	{
+		laneSpline_->RemoveSplinePoint(laneSpline_->GetNumberOfSplinePoints() - 2);
+	}
+}
+
+void AOpenDriveRoadEd::SetArrowMeshes(USplineComponent* laneSpline_, TObjectPtr<UStaticMesh> mesh, bool isJunction)
+{
+	for (int i = 1; i < laneSpline_->GetNumberOfSplinePoints() - 1; i += 2)
+	{
+		UStaticMeshComponent* newStaticMesh = NewObject<UStaticMeshComponent>(this);
+		newStaticMesh->SetupAttachment(RootComponent);
+		newStaticMesh->SetMobility(EComponentMobility::Movable);
+		newStaticMesh->SetStaticMesh(mesh);
+		newStaticMesh->SetWorldLocation(laneSpline_->GetWorldLocationAtSplinePoint(i) + FVector(0.f,0.f,5.0f));
+		FRotator rotation = laneSpline_->GetRotationAtSplinePoint(i, ESplineCoordinateSpace::World);
+		rotation.Yaw += _roadDirection == -1 ? FMath::RadiansToDegrees(PI) : 0.f;
+		newStaticMesh->SetWorldRotation(rotation);
+
+		if (isJunction == true)
+		{
+			newStaticMesh->SetWorldScale3D(newStaticMesh->GetComponentScale() / 2);
+		}
+		else
+		{
+			newStaticMesh->SetVisibility(false);
```

Comment:
@N3zMa1 Please provide some feedback on that ;)

Reply:
I've putted directional arrows on every lane. By default, they are visible on junctions, but hidden on other lanes to avoid having to much stuffs in the viewport. 

However if you want to check the roads' directions, those arrows can be shown by ticking a small box in the editor widget.